learntools/python/ex3.py

```diff
@@ -209,6 +209,7 @@ class OneTopping(FunctionProblem):
     ]
 
 class BlackJackProblem(CodingProblem):
+    _counts_for_points = False
```

Comment:
Hm, storing that kind of dynamic information really feels like something we'd want Jamielytics for. (In a pinch, I'd even consider stuffing it into a different field like `trace` rather than `valueTowardsCompletion`. Either way, we're kind of abusing the schema, but at least that way it doesn't interfere with progress tracking.)

Reply:
I disagree that this is abusing the schema.  

This use case was specifically described as one of the reasons to switch to this schema.

See bullet 4 of my June 18 comment on https://github.com/Kaggle/kaggleazure/pull/2295

Reply:
Also, I think we want to be able to work with this data in an integrated way with everything else we do analytics on (e.g. answer whether people who did well on qualitative questions did well on other questions... skipped other questions... etc.) Because Jamielytics doesn't include user id's that would be pretty hard.

Reply:
Sorry... Thought I'd posted this before you took a look.  meant to include this when first commenting... but copying it here now:

---

Since we already have a concept of point_value, one way I could see this working is giving each problem a `points_multiplier` property set to 1 by default.

The problem author could reset `points_multiplier` as part of `check`, creating a score that's continuous.  Then the line above becomes
kwargs['valueTowardsCompletion'] = self.problem.point_value * self.points_multiplier

Maybe there's a better process for how `points_multiplier` is updated. But I don't think it requires a breaking change to the current abstractions.

Reply:
> Maybe there's a better process for how points_multiplier is updated. But I don't think it requires a breaking change to the current abstractions.

For me the disconnect is that I'm imagining the intended purposes of the ValueTowardsCompletion field is determining whether a user has completed an exercise (i.e. whether they get a checkbox, or how much to fill in their progress bar). 

It sounds like what you're describing with the ticket pricing example is about collecting some data that's interesting from an analytics point of view, but not intended to be used to determine whether an exercise is done.

(I suppose you could say that, for this question, the target MSE is 80. So if your model has a MSE of 160, you're 50% done. But would you actually want to surface it that way in the UI? I think this would be especially confusing if this kind of scoring is combined with other questions in the same exercise. If I do really well on question 1 and do 4x better than the threshold, does that result in me getting a checkmark for the whole exercise, without touching questions 2, 3, and 4?)

Going back to that 4th bullet you referred to in the earlier PR, I think you kind of made the same point about UX of showing non-binary scores:

> Option for LearnTools to provide score on an exercise that is a function of how good a user's model is, rather than purely binary. **We likely won't do option 4 in the near term since it opens up a UX question of how to surface the requirements for a check to the user.**

(Emphasis mine)

Given that we're unlikely to do it in the near term, I'd say we should hold off on implementing it in LearnTools for now. If we decide we want it later, we can always add it.